x-pack/plugins/alerting/public/pages/maintenance_windows/components/maintenance_windows_list.tsx

```diff
@@ -108,38 +99,44 @@ const search: { filters: SearchFilterConfig[] } = {
 };
 
 export const MaintenanceWindowsList = React.memo<MaintenanceWindowsListProps>(
-  ({ loading, items }) => {
+  ({ loading, items, refreshData }) => {
+    const { euiTheme } = useEuiTheme();
     const { navigateToEditMaintenanceWindows } = useEditMaintenanceWindowsNavigation();
-    const warningBackgroundColor = useEuiBackgroundColor('warning');
-    const subduedBackgroundColor = useEuiBackgroundColor('subdued');
+    const { mutate: finishMaintenanceWindow } = useFinishMaintenanceWindow();
+    const { mutate: archiveMaintenanceWindow } = useArchiveMaintenanceWindow();
+    const { mutate: finishAndArchiveMaintenanceWindow } = useFinishAndArchiveMaintenanceWindow();
+
     const tableCss = useMemo(() => {
       return css`
         .euiTableRow {
           &.running {
-            background-color: ${warningBackgroundColor};
-          }
-
-          &.archived {
-            background-color: ${subduedBackgroundColor};
+            background-color: ${euiTheme.colors.highlight};
           }
         }
       `;
-    }, [warningBackgroundColor, subduedBackgroundColor]);
+    }, [euiTheme.colors.highlight]);
 
     const actions: Array<EuiBasicTableColumn<MaintenanceWindowFindResponse>> = [
       {
         name: '',
-        actions: [
-          {
-            name: i18n.TABLE_ACTION_EDIT,
-            isPrimary: true,
-            description: 'Edit maintenance window',
-            icon: 'pencil',
-            type: 'icon',
-            onClick: (mw: MaintenanceWindowFindResponse) => navigateToEditMaintenanceWindows(mw.id),
-            'data-test-subj': 'action-edit',
-          },
-        ],
+        render: ({ status, id }: { status: MaintenanceWindowStatus; id: string }) => {
+          return (
+            <TableActionsPopover
+              status={status}
+              onEdit={() => navigateToEditMaintenanceWindows(id)}
+              onCancel={() => finishMaintenanceWindow(id, { onSuccess: () => refreshData() })}
+              onArchive={(archive: boolean) =>
+                archiveMaintenanceWindow(
+                  { maintenanceWindowId: id, archive },
+                  { onSuccess: () => refreshData() }
+                )
+              }
+              onCancelAndArchive={() =>
+                finishAndArchiveMaintenanceWindow(id, { onSuccess: () => refreshData() })
+              }
```

Comment:
nitpick: Ideailly we should `useCallback` on these functions to help with performance :slightly_smiling_face:

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/155201/commits/9d6e6730e18bd54c5bece8e97a0c801d232edd3c